sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp

```diff
@@ -2,12 +2,130 @@
+#include "third_party/skia/include/core/SkTypeface.h"
+#include "third_party/skia/include/ports/SkFontMgr.h"
 
 namespace blink {
+namespace {
+
+uint32_t ToMojoWeight(FontWeight weight) {
+  switch (weight) {
+    case FontWeight100:
+      return 100;
+    case FontWeight200:
+      return 200;
+    case FontWeight300:
+      return 300;
+    case FontWeight400:
+      return 400;
+    case FontWeight500:
+      return 500;
+    case FontWeight600:
+      return 600;
+    case FontWeight700:
+      return 700;
+    case FontWeight800:
+      return 800;
+    case FontWeight900:
+      return 900;
+  }
+  ASSERT_NOT_REACHED();
+  return 400;
+}
+
+mojo::FontSlant ToMojoSlant(FontStyle style) {
+  switch (style) {
+    case FontStyleNormal:
+      return mojo::FontSlant::UPRIGHT;
+    case FontStyleItalic:
+      return mojo::FontSlant::ITALIC;
+  }
+  ASSERT_NOT_REACHED();
+  return mojo::FontSlant::UPRIGHT;
+}
+
+void UnmapMemory(const void* buffer, void* context) {
+  mx_process_unmap_vm(0, reinterpret_cast<uintptr_t>(buffer), 0);
+}
+
+sk_sp<SkData> MakeSkDataFromVMO(mx_handle_t vmo) {
+  uint64_t size = 0;
+  mx_status_t status = mx_vmo_get_size(vmo, &size);
+  if (status < 0 || size > std::numeric_limits<mx_size_t>::max())
+    return nullptr;
+  uintptr_t buffer = 0;
+  status = mx_process_vm_map(0, vmo, 0, size, &buffer, MX_VM_FLAG_PERM_READ);
+  if (status < 0)
+    return nullptr;
+  return SkData::MakeWithProc(reinterpret_cast<void*>(buffer), size,
+                              UnmapMemory, nullptr);
+}
+
+mojo::FontProviderPtr* g_font_provider = nullptr;
+
+mojo::FontProviderPtr& GetFontProvider() {
+  FTL_CHECK(g_font_provider);
+  return *g_font_provider;
+}
+
+}  // namespace
+
+void SetFontProvider(mojo::FontProviderPtr provider) {
+  FTL_CHECK(!g_font_provider);
+  g_font_provider = new mojo::FontProviderPtr;
+  *g_font_provider = std::move(provider);
+}
 
 void FontCache::getFontForCharacter(UChar32 c,
                                     const char* preferredLocale,
                                     PlatformFallbackFont* font) {}
 
+sk_sp<SkTypeface> FontCache::createTypeface(
+    const FontDescription& fontDescription,
+    const FontFaceCreationParams& creationParams,
+    CString& name) {
+  AtomicString family = creationParams.family();
+
+  if (family.isEmpty()) {
+    name = getFallbackFontFamily(fontDescription).string().utf8();
+  } else {
+    name = family.utf8();
+  }
+
+  auto request = mojo::FontRequest::New();
+  request->family = name.data();
+  request->weight = ToMojoWeight(fontDescription.weight());
+  request->width = static_cast<uint32_t>(fontDescription.stretch());
+  request->slant = ToMojoSlant(fontDescription.style());
+
+  mojo::FontResponsePtr response;
+  auto& font_provider = GetFontProvider();
+  font_provider->GetFont(
+      std::move(request),
+      [&response](mojo::FontResponsePtr r) { response = std::move(r); });
+  font_provider.WaitForIncomingResponse();
```

Comment:
They're all created on the IU thread.  The Mojo bindings will helpfully assert if you try to use an InterfacePtr from multiple threads.